portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +33,39 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Fetches greeting from the server and adds it to the DOM.
+ */
+function getGreeting() {
+  console.log('Fetching greeting.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addGreetingToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addGreetingToDom() function.
+  textPromise.then(addGreetingToDom);
+}
+
+/** Adds greeting to the DOM. */
+function addGreetingToDom(greeting) {
+  console.log('Adding greeting to dom: ' + greeting);
+  const greetingContainer = document.getElementById('greeting-container');
+  greetingContainer.innerHTML = greeting;
+}
```

Comment:
This code is no longer being used, right? consider removing it to avoid confusion.

Reply:
Yes that is correct! I will remove it.